viewing_party/party.py

```diff
@@ -1,23 +1,125 @@
 # ------------- WAVE 1 --------------------
 
 def create_movie(title, genre, rating):
-    pass
+    if title and genre and rating:
+        movie_dict = {"title": title,
+        "genre": genre,
+        "rating": rating
+        }
+        return movie_dict
+    else:
+        return None
+
+def add_to_watched(user_data, movie):
+    watched_list = []
+    watched_list.append(movie)
+    user_data["watched"] = watched_list
+    return user_data
+
+def add_to_watchlist(user_data, movie):
+    watchlist = []
+    watchlist.append(movie)
+    user_data["watchlist"] = watchlist
+    return user_data
+
+def watch_movie(user_data, title):
+    for movie in user_data["watchlist"]:
+        if title == movie["title"]:
+            user_data["watchlist"].remove(movie)
+            user_data["watched"].append(movie)
+    return user_data
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
 
+def get_watched_avg_rating(user_data):
+    total = 0.0
+    movies = 0
+    for movie in user_data["watched"]:
+        total += movie["rating"]
+        movies += 1
+    if movies == 0:
+        avg = 0.0
+    else:
+        avg = total / movies
+    return avg
+
+def get_most_watched_genre(user_data):
+    count = 0
+    max_count = 0
+    max_genre = ""
+    for movie in user_data["watched"]:
+        genre = movie["genre"]
+        for movie in user_data["watched"]:
+            if genre in movie["genre"]:
+                count += 1
+        if count > max_count:
+            max_genre = genre
+            max_count = count
+        count = 0
+    if max_count == 0:
+        return None
+    else:
+        return max_genre    

```

Comment:
Solid solution!  It is O(n^2), there is an O(n) solution that uses a dictionary.